File: backend/entity/address.go
package entity

import "gorm.io/gorm"

type Address struct {
	gorm.Model
	HouseNo     string `json:"house_no"`
	VillageNo   string `json:"village_no"`
	Village     string `json:"village"`
	Alley       string `json:"alley"`
	Road        string `json:"road"`
	SubDistrict string `json:"sub_district"`
	District    string `json:"district"`
	Province    string `json:"province"`
	ZipCode     string `json:"zip_code"`

	// One-to-one relationship with Student
	//StudentID string
	StudentID uint      `json:"student_id"`
	Student   *Students `gorm:"foreignKey: student_id;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;" json:"student"`
}

File: frontend/src/interfaces/Address.ts
export interface AddressInterface{
  ID?: number;
  house_no?:  string;    
    village_no?: string;    
    village?:  string;   
    alley?:     string;    
    road?:  string; 
    sub_district?:  string;   
    district?: string;
    province?: string;
    zip_code?: string;
  StudentID?: string; // เชื่อมโยงกับ Student
}
